Wasm-container/README.md

```diff
@@ -0,0 +1,509 @@
+calls. Supporting Web Assembly UDxes lets us support unfenced UDxes in
+any language that can compile to Wasm, confident that the user-defined
+code cannot corrupt Vertica's data structures.
+
+For details about developing Vertica UDxes, see [Extending Vertica](https://www.vertica.com/docs/latest/HTML/Content/Authoring/ExtendingVertica/ExtendingVertica.htm).
+
+This container just packages WebAssembly tools for generating .wasm files in C and Rust, though we anticipate adding other languages (particularly Golang).
+
+## Prerequisites
+- [Docker Desktop](https://www.docker.com/get-started) or [Docker Engine](https://docs.docker.com/engine/install/).
+- `make`: a program for building programs.
+- A copy of a Vertica `.deb` release file.
+
+# Supported Platforms
+
+We use an Ubuntu container primarily because we found it easier to install the WebAssembly tools we need in Ubuntu.
+
+# Overview
+
+This container packages:
+
+- the Vertica SDK for developing UDxes
+- a Vertica runtime to use to test your UDx
+- `llvm` --- the `llvm` compiler components capable of creating WebAssembly output 
+- `emscripten` --- a compiler package for compiling to WebAssembly
+- `wasmer` --- WebAssembly tools
+- `rust` --- a version of the Rust compiler that can compile to WebAssembly
+- some scripts to make it easier to use the contents of this container
+- `/usr/WebAssembly/vertica` --- a collection of functions exploiting WebAssembly to make unfenced Vertica UDxes in C and Rust.  This is a work in progress.  Also included are some "scaffolding" --- functions written to explore features of the C Wasm API.
+
+# Building the container
+
+The container is built using the command `make`.  `make` will create a Docker image called `wasm-playground:ubuntu`.
+The Makefile requires that you store a Vertica DEB file in the top-level directory of your cloned repository. The container inherits the privileges and user ID from the user executing the container.
+
+## Build variables
+
+You can include build variables in the build process to customize the
+container. The following table describes the available variables: 
+
+| Name                      | Definition |
+|---------------------------|------------|
+| PACKAGE | When there is more than one Vertica DEB file in the top-level directory, this variable specifies which file to use in the build process. |
+| VERTICA_VERSION | The version number of the Vertica binary used in the build process. This value is optional for a [canonically-named Vertica binary](#building-with-a-canonically-named-vertica-binary).<br> You can use this variable to build containers for different Vertica versions. |
+
+For example, you might build multiple containers to develop UDxes for multiple Vertica versions. To help distinguish between containers, `VERTICA_VERSION` in the build command. If you set `VERTICA_VERSION=11.0.0-0`, the full container specification is `vwasmsdk:ubuntu-11.0.0-0`.
```

Comment:
"To help distinguish between containers, VERTICA_VERSION in the build command" -- missing word?  "...containser, set VERTICA_VERSION..."?